File: JS/1704.determine_if_string_halves_are_alike [1].js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    function isVowel(ch){
        ch = ch.toLowerCase()
        return ch === 'a' || ch === 'e' || ch === 'i' || ch === 'o' || ch === 'u'
    }

    const n = s.length

    let leftVowelCount = 0;
    for(let i=0; i<n/2; i++)
        if(isVowel(s[i]))
            leftVowelCount++
    
    let rightVowelCount = 0;
    for(let i=n/2; i<n; i++)
        if(isVowel(s[i]))
            rightVowelCount++

    return leftVowelCount === rightVowelCount
};

File: CPP [1501-2000]/1704.determine-if-string-halves-are-alike [1].cpp
class Solution {
private:
    bool isVowel(char ch){
        ch = tolower(ch);

        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }

public:
    // T.C.=O(n), S.C.=O(1)
    bool halvesAreAlike(string s) {
        int n=s.length();

        int leftVowelCount = 0;
        for(int i=0; i<n/2; i++)
            if(isVowel(s[i]))
                leftVowelCount++;
        
        int rightVowelCount = 0;
        for(int i=n/2; i<n; i++)
            if(isVowel(s[i]))
                rightVowelCount++;

        return leftVowelCount == rightVowelCount;
    }
};